pkg/service/pipeline.go

```diff
@@ -866,38 +864,31 @@ func (s *service) preTriggerPipeline(ctx context.Context, ns resource.Namespace,
 
 		// TODO: remove these conversions after the blob storage is fully rolled out
 		for k := range m {
-			switch s := m[k].(type) {
+			switch str := m[k].(type) {
 			case string:
 				if formatMap[k] != "string" && formatMap[k] != "number" && formatMap[k] != "boolean" && formatMap[k] != "json" {
 					// Skip the base64 decoding if the string is a URL
-					if strings.HasPrefix(s, "http://") || strings.HasPrefix(s, "https://") {
+					if strings.HasPrefix(str, "http://") || strings.HasPrefix(str, "https://") {
 						continue
 					}
-					if !strings.HasPrefix(s, "data:") {
-						b, err := base64.StdEncoding.DecodeString(s)
-						if err != nil {
-							return fmt.Errorf("can not decode file %s, %s", formatMap[k], s)
-						}
-						mimeType := strings.Split(mimetype.Detect(b).String(), ";")[0]
-						vars.Fields[k] = structpb.NewStringValue(fmt.Sprintf("data:%s;base64,%s", mimeType, s))
+					downloadURL, err := s.uploadBlobAndGetDownloadURL(ctx, ns, str)
```

Comment:
❓ Is every string interpreted as a data blob (and therefore uploaded)? E.g. in a text generation chat, will the chat input be uploaded in this fashion?

Reply:
No, only when the instill format meet this condition.
`formatMap[k] != "string" && formatMap[k] != "number" && formatMap[k] != "boolean" && formatMap[k] != "json" {`

I will use another function to make it more readable.

e.g. to follow the [term here](https://www.instill.tech/docs/vdp/instill-format#unstructured-data-types)
```
func isUnstructuredData(format string) {
   ...
}
```